crates/alerter/src/main.rs

```diff
@@ -196,10 +230,16 @@ impl Alerter {
                 (order, existing_time)
             })
             .collect::<Vec<_>>();
+
         tracing::debug!("found {} open orders", orders.len());
+
         std::mem::swap(&mut self.open_orders, &mut orders);
         // Keep only orders that were open last update and are not open this update.
-        orders.retain(|order| !self.open_orders.contains(order));
+        orders.retain(|(order, _)| {
+            self.open_orders
+                .iter()
+                .all(|(open_order, _)| open_order.uid != order.uid)
```

Comment:
not that it matters, but I have to point out accidentally quadratic complexity when I see it 🙈.

Reply:
🙈.

This needed to change because of `surplus_fee` semantics means that you can't use direct order comparison.

It was also quadratic before, so I didn't bother changing it 😛.

Reply:
https://github.com/cowprotocol/services/pull/1015